### ISSUE TITLE
Simplify query

### ISSUE THREAD

**Issue by placintaalexandru** [ArroyoSystems/arroyo#797]
I have a json that can be retrieved from [here](https://hermes.pyth.network/v2/updates/price/latest?ids%5B%5D=0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43&ids%5B%5D=0xc96458d393fe9deb7a7d63a0ac41e2898a67a7750dbd166673279e06c868df0a).

This is the JSON schema I use:
```json
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "title": "Generated schema for Root",
    "type": "object",
    "properties": {
        "binary": {
            "type": "object",
            "properties": {
                "encoding": {
                    "type": "string"
                },
                "data": {
                    "type": "array",
                    "items": {
                        "type": "string"
                    }
                }
            },
            "required": [
                "encoding",
                "data"
            ]
        },
        "parsed": {
            "type": "array",
            "items": {
                "type": "object",
                "properties": {
                    "id": {
                        "type": "string"
                    },
                    "price": {
                        "type": "object",
                        "properties": {
                            "price": {
                                "type": "string"
                            },
                            "conf": {
                                "type": "string"
                            },
                            "expo": {
                                "type": "number"
                            },
                            "publish_time": {
                                "type": "number"
                            }
                        },
                        "required": [
                            "price",
                            "conf",
                            "expo",
                            "publish_time"
                        ]
                    },
                    "ema_price": {
                        "type": "object",
                        "properties": {
                            "price": {
                                "type": "string"
                            },
                            "conf": {
                                "type": "string"
                            },
                            "expo": {
                                "type": "number"
                            },
                            "publish_time": {
                                "type": "number"
                            }
                        },
                        "required": [
                            "price",
                            "conf",
                            "expo",
                            "publish_time"
                        ]
                    },
                    "metadata": {
                        "type": "object",
                        "properties": {
                            "slot": {
                                "type": "number"
                            },
                            "proof_available_time": {
                                "type": "number"
                            },
                            "prev_publish_time": {
                                "type": "number"
                            }
                        },
                        "required": [
                            "slot",
                            "proof_available_time",
                            "prev_publish_time"
                        ]
                    }
                },
                "required": [
                    "id",
                    "price",
                    "ema_price",
                    "metadata"
                ]
            }
        }
    },
    "required": [
        "binary",
        "parsed"
    ]
}
```

My question is how I can access the price field that corresponds to this json path query `$.parsed[0].price.price`.

I use the following query, that works, but I find it quite complex for such a trivial task:
```
SELECT parsed_price.price
FROM (SELECT parsed_field.price AS parsed_price
    FROM (SELECT parsed[1] AS parsed_field FROM test_solana));
```

Basically I need a `select` statement every time I need to go deep inside the JSON object
